sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/implementation/Batcher.java

```diff
@@ -139,16 +139,18 @@ private LogsIngestionRequest nextInternal() throws IOException {
 
     private LogsIngestionRequest createRequest(boolean last) throws IOException {
         ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-        JsonGenerator generator = JsonFactory.builder().build().createGenerator(byteArrayOutputStream);
+        JsonWriter writer = JsonProviders.createWriter(byteArrayOutputStream);
         try {
-            generator.writeStartArray();
-            generator.writeRaw(serializedLogs.stream().collect(Collectors.joining(",")));
-            generator.writeEndArray();
-            generator.close();
-
+            writer.writeStartArray();
+            for (String log : serializedLogs) {
+                writer.writeRawValue(log);
+            }
+            writer.writeEndArray();
+            writer.close();
```

Comment:
Still need to add `JsonWriter` to the try-with-resource and change this call to `flush()`